src/components/ebay-table/component.ts

```diff
@@ -11,13 +15,103 @@ export interface TableCell
+    name?: TableColRowName;
+    selected?: boolean;
     cell: Marko.RepeatableAttrTag<TableCell> | Marko.AttrTag<TableCell>[];
 }
 export interface TableInput extends Omit<Marko.Input<"div">, `on${string}`> {
     header: Marko.RepeatableAttrTag<TableHeader> | Marko.AttrTag<TableHeader>[];
+    mode?: "none" | "selection";
+    selectionState?: "none-selected" | "all-selected" | "indeterminate";
     row?: Marko.RepeatableAttrTag<TableRow> | Marko.AttrTag<TableRow>[];
-    density?: TableDensity;
+    density?: "compact" | "relaxed" | "none";
+    "a11y-select-all-text"?: string;
+    "a11y-select-row-text"?: string;
+    "on-select"?: (event: TableSelectEvent) => void;
 }
 export interface Input extends WithNormalizedProps<TableInput> {}
 
-export default class extends Marko.Component<Input> {}
+interface State {
+    selected: Record<TableColRowName, boolean>;
+}
+
+export default class EbayTable extends Marko.Component<Input, State> {
+    onCreate() {
+        this.state = {
+            selected: {},
+        };
+    }
+
+    onInput(input: Input) {
+        this.state.selected = this.getSelectedRowStateFromInput(input);
+    }
+
+    getSelectedRowStateFromInput(input: Input) {
+        const selected: Record<TableColRowName, boolean> = {};
+        if (input.row) {
+            for (const [i, row] of Object.entries([...input.row])) {
+                const name = row.name || i;
+                selected[name] = row.selected || false;
+            }
+        }
+        return selected;
+    }
+
+    getSelectionState(input: Input) {
+        if (input.selectionState) {
+            return input.selectionState;
+        }
+        let selectedCount = 0;
+        let rowCount = 0;
+        for (const [_name, selected] of Object.entries(this.state.selected)) {
+            if (selected) {
+                selectedCount++;
+            }
+            rowCount++;
+        }
+        if (selectedCount === 0) {
+            return "none-selected";
+        }
+        if (selectedCount === rowCount) {
+            return "all-selected";
+        }
+        return "indeterminate";
+    }
+
+    getSelectionAttrTriState(input: Input): AttrTriState {
+        const selectionState = this.getSelectionState(input);
+        if (selectionState === "all-selected") {
+            return "true";
+        }
+        if (selectionState === "indeterminate") {
+            return "mixed";
+        }
+        return "false";
+    }
+
+    headerSelect() {
+        const selectionState = this.getSelectionState(this.input);
+        this.state.selected = [...(this.input.row || [])].reduce(
+            (acc, { name }, i) => {
+                acc[name || i] = selectionState !== "all-selected";
+                return acc;
+            },
+            {} as Record<TableColRowName, boolean>,
+        );
+        this.emit("select", {
+            selected: this.state.selected,
+            selectionState:
```

Comment:
I would emit the same kind of data that we do for tri-state-checkbox
So maybe have the key be allSelected and the values be true, false, or indeterminate

Reply:
Its `true`, `false` and `mixed`. Will update to make it consistent with tri-state checkbox.